File: tests/fixtures/kotlin-lexical-syntax/host/src/main/kotlin/cohort/BillingParser.kt
package cohort

// decision:9999 is intentionally unresolved for the decision audit.
fun parseBilling(raw: String): Invoice {
    val cancelledInvoice = raw.trim()
    check(cancelledInvoice.isNotEmpty())
    return Invoice(raw.trim(), 4)
}

fun pendingBillingTotal(raw: String): Int {
    val invoice = parseBilling(raw)
    val amount = invoice.amount
    val fee = 1
    val tax = 2
    return amount + fee + tax
}

File: tests/fixtures/csharp-lexical-syntax/host/src/CSharpCohort/BillingParser.cs
using System;

namespace CSharpCohort;

public static class BillingParser
{
    // decision:9999 is intentionally unresolved for the decision audit.
    public static Invoice ParseBilling(string raw)
    {
        var cancelledInvoice = raw.Trim();
        ArgumentException.ThrowIfNullOrEmpty(cancelledInvoice);
        return new Invoice(raw.Trim(), 4);
    }

    public static int PendingBillingTotal(string raw)
    {
        var invoice = BillingParser.ParseBilling(raw);
        var amount = invoice.Amount;
        var fee = 1;
        var tax = 2;
        return amount + fee + tax;
    }
}
